src/bpftrace.cpp

```diff
@@ -48,53 +48,67 @@ namespace bpftrace {
 DebugLevel bt_debug = DebugLevel::kNone;
 bool bt_verbose = false;
 volatile sig_atomic_t BPFtrace::exitsig_recv = false;
+const int FMT_BUF_SZ = 512;
 
-int format(char * s, size_t n, const char * fmt, std::vector<std::unique_ptr<IPrintable>> &args) {
+std::string format(std::string fmt,
+                   std::vector<std::unique_ptr<IPrintable>> &args)
+{
+  std::string retstr;
+  int n = FMT_BUF_SZ;
+  auto buffer = std::vector<char>(n);
+  char *str = buffer.data();
```

Comment:
Drop this temporary too. It's more clear to the reader when they see `buffer.data()` than `str`.